test/xspec-variable.xspec

```diff
@@ -5,29 +5,212 @@
 <!--  Tags:                                                                -->
 <!--    Copyright (c) 2010 Jeni Tennison (see end of file.)                -->
 <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
-<t:description xmlns:t="http://www.jenitennison.com/xslt/xspec" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:my="http://example.org/ns/my" query="http://example.org/ns/my" query-at="xspec-tested.xql" stylesheet="xspec-tested.xsl">
+
+<!-- TODO: Make the following changes after other code gets merged onto master branch.
+  - Test that variable/@name should be EQNAME and should be added to EQName test. See
+    test/xspec-eqname.xspec on AirQuick's branch.
+  - Remove this file from schema test exclusion list. See https://github.com/xspec/xspec/pull/363.
+  - To better test different namespace prefixes, consider updating after
+    https://github.com/xspec/xspec/commit/b1e6d150644ea323252ea3fd9f5addbdde1da40a
+    is integrated. See files named test/xspec-prefix-conflict*.* on AirQuick's branch.
+  - Restore "Text directly in XSpec variable" scenario. XQuery $myspace1 needs x:text
+    support in schema and compiler.
+  - Move "@select and element content" scenario from xspec-variable_stylesheet.xspec to this file.
+    Need certain changes in generate-query-helper.xsl; see <xsl:template match="*" mode="test:generate-variable-declarations">.
+  -->
+
+<t:description xmlns:t="http://www.jenitennison.com/xslt/xspec" xmlns:x="http://www.jenitennison.com/xslt/xspec" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:my="http://example.org/ns/my" stylesheet="xspec-variable-tested.xsl" query="http://example.org/ns/my" query-at="xspec-variable-tested.xql">
   <t:variable name="test-data" href="xspec-variable.xml"/>
   <t:variable name="test-3" select="$test-data/*/test[xs:integer(@num) eq 3]"/>
-  <t:scenario label="call global var">
-    <t:call function="my:square">
-      <t:param select="$test-3/@num"/>
-    </t:call>
-    <t:expect label="the result" test="$x:result eq $test-3/xs:integer(@value)"/>
+
+  <t:scenario label="Define variables in different ways">
+    <t:scenario label="@select only">
+      <t:variable name="select-only" select="'y'"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{$select-only}"/></t:param>
```

Comment:
`<foo role="{$select-only}"/>`: This attribute-value-template-like expression may work now, but I think the current behavior is a bug and fixed it in #454. Do you usually rely on the current behavior when you write XSpec?

Reply:
Thanks for letting me know. I'll change the handful of scenarios that I wrote using this pattern. Had I known the current behavior was considered a bug, I would not have used or relied on it.

Reply:
Oops! Sorry, please continue using `<foo role="{$select-only}"/>`.
Looking into the history further, I found that the current behavior was actually introduced as a feature, although it was not discussed and not documented and not tested at all and many seemingly related bugs are scattered. I'll revise #454.

Reply:
Sorry for taking your time, and thanks for bringing this to attention.
I updated [Wiki](https://github.com/xspec/xspec/wiki/Writing-Scenarios/_compare/c2c62195370a69fe91d263a86d17c796d08984a3...0839488d590a01c9e0f2db3543f1b8ce1d969a91) to clarify the behavior, based on expath/xspec#9 and [a past thread](https://groups.google.com/d/msg/xspec-users/PstZ7Yaois8/MIrEA9o7lGMJ).

Reply:
No problem! :) Thanks for checking back on this and clarifying things.